tests/main/services-restart/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Check that snap restarts affects only running services
+
+execute: |
+    echo "Installing the service snap"
+
+    # we are using systemd-notify indicate the service is active, this is
+    # currently not allowed by daemon-notify interface, so we may as well just
+    # install in devmode
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-service-restart --devmode
+
+    echo "We can see all services running"
+    for id in 1 2 3 4; do
+        systemctl status snap.test-snapd-service-restart.service$id | MATCH "running"
+    done
+
+    echo "Stopping services 3 and 4"
+    snap stop test-snapd-service-restart.service3
+    snap stop test-snapd-service-restart.service4
+
+    echo "Disabling services 1 and 3"
+    systemctl disable snap.test-snapd-service-restart.service1
+    systemctl disable snap.test-snapd-service-restart.service3
```

Comment:
We should probably grab the `ExecMainStartTimestampMonotonic` property of relevant services, and then after `snap restart` compare that it has not changed.

Reply:
Wait, do you meant to say "...compare that it **has** changed"? Or did I misunderstand your suggestion?

Reply:
I meant it should change for the ones we expected to see restarted, and should stay the same for ones that were inactive